File: src/main/webapp/WEB-INF/src/js/app/administration/groups/GroupService.js
module.exports = function ($http) {
    var _this = this;

    _this.findPage = function (subjectUuid) {
        var params = {
            size: 10000000,
            page: 0,
            subjectUuid: subjectUuid || null
        };
        return $http.get('/api/groups', {params: params}).then(function (response) {
            return response.data;
        });
    };

    _this.findOne = function (uuid) {
        var params = {};
        return $http.get('/api/groups/' + uuid, {params: params}).then(function (response) {
            return response.data;
        });
    };

    _this.create = function (subject) {
        return $http.post('/api/groups/', subject).then(function (response) {
            return response.data;
        });
    };

    _this.update = function (subject) {
        return $http.put('/api/groups/' + subject.uuid, subject).then(function (response) {
            return response.data;
        })
    };

    _this.delete = function (subject) {
        return $http.delete('/api/groups/' + subject.uuid);
    }
};

File: src/main/java/uz/com/journal/app/group/GroupService.java
package uz.com.journal.app.group;


import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import uz.com.journal.core.NonDeletableService;
import uz.com.platform.app.users.User;

public interface GroupService extends NonDeletableService<Group> {
    Page<Group> findPage(GroupFetch fetch, GroupCriteria criteria, Pageable pageable, User currentUser);

    Group findOne(String uuid, GroupFetch fetch, User currentUser);

    Group create(GroupItem item, User currentUser);

    Group update(String uuid, GroupItem item, User currentUser);
}
